### PR TITLE
refactor: upgrade styling dependencies

### PR DESCRIPTION
**Pull request by damianstasik**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description
I updated `css-loader` and `sass-loader` to latest version, replaced `node-sass` with `sass` as the former is deprecated and fixed a few things that broke due the switch.

As a bonus we have one less critical vulnerability and a lot less dependencies.

Related issue: #1902

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by mtrezza** [comment]
As usual, when this is ready for review please request a review.

File: package.json [hunk 1]
```diff
@@ -102,14 +102,14 @@
...
     "marked": "4.0.10",
-    "node-sass": "7.0.3",
     "null-loader": "4.0.1",
     "path-to-regexp": "3.2.0",
     "puppeteer": "18.0.5",
     "react-test-renderer": "16.13.1",
     "request": "2.88.2",
     "request-promise": "4.2.5",
-    "sass-loader": "12.3.0",
+    "sass": "^1.55.0",
```
Comment (mtrezza) [inline package.json]:
```suggestion
    "sass": "1.55.0",
```

File: package.json [hunk 2]
```diff
@@ -102,14 +102,14 @@
     "jest-environment-jsdom": "^29.1.2",
```
Comment (mtrezza) [inline package.json]:
```suggestion
    "jest-environment-jsdom": "29.1.2",
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Review by mtrezza — APPROVED** [review]
Looks good! Waiting for CI...

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-alpha.4](https://github.com/parse-community/parse-dashboard/releases/tag/5.0.0-alpha.4)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.0.0)

**Pull request merged**